keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -0,0 +1,343 @@
+      * [Implementation History](#implementation-history)
+      * [Drawbacks](#drawbacks)
+      * [Alternatives](#alternatives)
+      * [Infrastructure Needed](#infrastructure-needed)
+
+
+## Release Signoff Checklist
+
+**ACTION REQUIRED:** In order to merge code into a release, there must be an issue in [kubernetes/enhancements] referencing this KEP and targeting a release milestone **before [Enhancement Freeze](https://github.com/kubernetes/sig-release/tree/master/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core Kubernetes i.e., [kubernetes/kubernetes], we require the following Release Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+This KEP generalizes the existing max-in-flight request handler in the
+apiserver to make more distinctions among requests and provide
+prioritization and fairness among the categories of requests.  An
+outline of the request handling in an apiserver can be found at
+https://speakerdeck.com/sttts/kubernetes-api-codebase-tour?slide=18 .
+
+## Motivation
+
+Today the apiserver has a simple mechanism for protectimg itself
+against CPU and memory overloads: max-in-flight limits for mutating
+and for readonly requests.  Apart from the distinction between
+mutating and readonly, no other distinctions are made among requests;
```

Comment:
There's a 3rd group (perhaps subgroup?) for long running, I thought.

Reply:
Yes and no.  The long-running requests are exempt from management by this handler; so you can count them or not, depending on how you prefer to count.